docs/extension-system/_index.md

```diff
@@ -0,0 +1,148 @@
+
+```ts
+// Install '@ownclouders/web-pkg' as a devDependency first (only relevant for types and autocompletion, dependency is already provided by the ownCloud Web at runtime). 
+import {
+  AppWrapperRoute,
+  ApplicationFileExtension,
+  defineWebApplication
+} from '@ownclouders/web-pkg'
+
+
+export default defineWebApplication({
+  setup({ applicationConfig }) {
+    // Here, you have access to the full injection context, meaning you can use all composable that we provide via web-pkg
+
+    // Needs to be unique within all installed extensions in any ownCloud web instance
+    // Should be short, unique and expressive as it gets prefixed on all routes within your application
+    const appId = 'your-extension' 
+
+    // See extensions section below
+    const extensions = [
+        ...
+    ]
+
+    // See details below
+    const navItems = [
+      ...
+    ]
+
+    // See details below
+    const routes = [
+        ...
+    ]
+
+    return {
+      appInfo: {
+        name: $gettext('Your extension'),
+        id: appId,
+        icon: 'aliens', // See https://owncloud.design/#/Design%20Tokens/IconList for available options
+      },
+      extensions,
+      navItems,
+      routes
+    }
+  }
+})
+```
+
+By defining an application via `defineWebApplication` you can provide the following:
+- `appInfo` - the application metadata, which is used to make the application available via the app switcher and the app registry
+- `navItems` - the statically defined navigation items for the left side bar. Only gets rendered when more than 1 navigation item exists at runtime. 
+Additional dynamic navigation items can be registered via the extension registry.
+- `routes` - the routes to the different views of your application. May be referenced within the `navItems`. Authentication requirements can be defined per item.
+- `extensions` - the extensions to be registered in the extension registry. For more details see the `Extensions` section below.
+
+### Extensions
+
+In contrast to applications, extensions usually have a rather small scope and dedicated functionality.
+
+#### Extension Registry
+
+The globally available extension registry provided by the ownCloud Web runtime can be used to both register and query extensions. All extensions
+which are being made available via an `app` get registered in the extension registry automatically. In your custom application code you can
+then query any of the available extensions by providing a `type` string and optionally a list of `scopes`. Throughout the ownCloud Web platform
+and most prominently also in the `files` app we have defined some extension points which automatically use certain extensions, see the 
+`Extension Points` section below.
+
+#### Extension Types
+
+Building an extension is limited to the extension types that are defined by the ownCloud Web extension system. See the full list of available extension types below.
```

Comment:
Why is it limited? You can build five apps that share their own extension types which are injected in one another

Reply:
It's not, you're right. Changed it to "these are the predefined ones and you can always define your own" basically.